aiida/engine/processes/workchains/workchain.py

```diff
@@ -121,24 +121,53 @@ def on_run(self):
         super().on_run()
         self.node.set_stepper_state_info(str(self._stepper))
 
+    def _resolve_nested_context(self, key: str) -> Tuple[AttributeDict, str]:
+        """
+        Returns a reference to a sub-dictionary of the context and the last key,
+        after resolving a potentially segmented key where required sub-dictionaries are created as needed.
+
+        :param key: A key into the context, where words before a dot are interpreted as a key for a sub-dictionary
+        """
+        ctx = self.ctx
+        ctx_path = key.split('.')
+
+        for path in ctx_path[:-1]:
+            try:
+                ctx = ctx[path]
+            except TypeError:
+                raise AssertionError(
+                    f'The sub-dictionary part `{path}` of the given nested key path `{key}` is already set to a list'
+                )
+            except KeyError:
+                ctx[path] = AttributeDict()  # create the sub-dict and update the context
+                ctx = ctx[path]
+
+        # Awaitables are also AttributeDicts, but we must NEVER attach other Awaitables to them
+        assert not isinstance(
+            ctx, (Awaitable, list)
+        ), f'The sub-dictionary path `{".".join(ctx_path[:-1])}` points to an Awaitable or a Sequence'
```

Comment:
Just like the other `AssertionError`, I think this should be a `ValueError` because it would be thrown in case of an inconsistent user provided error. A minor-nitpick on the side: you check for `list` but say `Sequence` in the help string, probably good to align these two.

Reply:
Sure, will do. I'll go toward `Sequence` I guess since that's the more general one.

Reply:
ok, the complete thing needed to change, done.